srv/salt/_modules/packagemanager.py

```diff
@@ -0,0 +1,274 @@
+from subprocess import Popen, PIPE
+from platform import linux_distribution
+import logging
+import os
+
+log = logging.getLogger(__name__)
+
+
+class PackageManager(object):
+
+    """
+    That is not the nativ salt module and is not ment to
+    replace it.
+    This module was created to react on packagemanagers
+    reboot advice.
+    """
```

Comment:
After looking at the code more, I do see why 'reinventing the wheel' is actually necessary here. Perhaps we continue as-is for now and make a note to push the functionality upstream into salt when we have cycles?